presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1104,34 +1107,32 @@ else if (criteria instanceof JoinOn) {
 
             for (Expression conjunct : ExpressionUtils.extractConjuncts((Expression) optimizedExpression)) {
                 conjunct = ExpressionUtils.normalize(conjunct);
```

Comment:
Yeah - right. There still can be sth like `NOT(NOT(a.x = b.x))`. But maybe iwe can assume that it so rare that we do not want to care about it. 

And if we still wanted to keep normalization but move it to `SimplifyExpression` rewrite - does what I wrote make sense? Do we want to add original join condition into `JoinNode`? Or you thought about another approach?